backend/src/services/cars/carsSearch.service.ts

```diff
@@ -0,0 +1,135 @@
+import { CarsSearchParams } from '@autoline/shared';
+import { AutoriaRequestParams } from '@common/types/cars/autoria_request_params';
+import { prisma } from '@data/prisma-client';
+
+const carsSearch = async (
+  data: CarsSearchParams,
+): Promise<AutoriaRequestParams> => {
+  const result: AutoriaRequestParams = { category_id: 1 };
+
+  if (data.brandId || data.modelId || data.bodyTypeId) {
+    const model_from_db = await prisma.model.findMany({
+      select: {
+        autoria_code: true,
+        body_type: {
+          select: {
+            autoria_code: true,
+          },
+        },
+        brand: {
+          select: {
+            autoria_code: true,
+          },
+        },
+      },
+      where: {
+        id: data.modelId,
+        body_type: {
+          id: data.bodyTypeId,
+        },
+        brand: {
+          id: {
+            in: data.brandId,
+          },
+        },
+      },
+    });
+
+    if (model_from_db?.length === 0) throw new Error('NotFoundError');
+
+    model_from_db.forEach((model, index) => {
+      Object.assign(result, {
+        [`body_type[${index}]`]: model.body_type?.autoria_code,
+        [`marka_id[${index}]`]: model.brand?.autoria_code,
+        [`model_id[${index}]`]: data.modelId ? model.autoria_code : undefined,
+        [`s_yers[${index}]`]: data.yearStart,
+        [`po_yers[${index}]`]: data.yearEnd,
+      });
+    });
```

Comment:
Could you explain what it does?

Reply:
It's just the way autoria asks to specify params. If you want to search multiply models or brends you need to add them this way `marka_id[0]=423&marka_id[1]=222` and so on. And if you want to search by years you need to duplicate it for every model/brand. That's actually what this code does